apis/events/src/event_traits.rs

```diff
@@ -177,6 +178,20 @@ impl EventSource {
         }
         self
     }
+
+    /// Add state change event specific data event source.
+    pub fn with_state_change_data(self, previous: &str, next: &str) -> Self {
```

Comment:
Probably here we should take String, reason being we're most likely converting a type to String, so better take String now to avoid 2 conversions!
```suggestion
    pub fn with_state_change_data(self, previous: String, next: String) -> Self {
```

Reply:
Changed to String.